src/NServiceBus.Core/Routing/DistributionStrategy.cs

```diff
@@ -24,6 +27,14 @@ protected DistributionStrategy(string endpoint, DistributionStrategyScope scope)
         public abstract string SelectReceiver(string[] receiverAddresses);
 
         /// <summary>
+        /// Selects a destination instance for a message from all known addresses of a logical endpoint.
+        /// </summary>
+        public virtual Task<string> SelectReceiver(string[] receiverAddresses, IOutgoingContext outgoingContext)
```

Comment:
Yes definitely we should always favour abstract classes over interfaces if we can because from a semver perspective it is not possible to obsolete an interface method because but we can properly introduce new methods on an abstract class as an additive change without breaking downstreams